src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -965,9 +975,18 @@ DQUnpackDictGatherCWu::DQUnpackDictGatherCWu(Context::Ref ctx) {
 
         // Strip down the DQ subgraph, replace the original Q-ed closure tensor with unpacked fp16
         auto new_wi = ctx.get().unpack(matched_qweight, matched_qzerop, matched_qcoeff, ov::element::f16);
-        auto gather_c = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{}, 0);
-        auto new_g = std::make_shared<ov::op::v8::Gather>(new_wi, matched_out_ids, gather_c);
+        auto w_shape = matched_node_qweight->get_shape();
+        auto new_w_shape = new_wi->get_shape();
+        std::shared_ptr<ov::op::v1::Reshape> opt_reshape = nullptr;
+        if (new_w_shape.size() == 2 && w_shape.size() == 3) {
+            NPUW_ASSERT(new_w_shape[0] == w_shape[0] && w_shape[1] * w_shape[2] == new_w_shape[1]);
+            auto new_const = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{3}, w_shape);
+            opt_reshape = std::make_shared<ov::op::v1::Reshape>(new_wi, new_const, false);
+        }
```

Comment:
```suggestion
            gather_in = std::make_shared<ov::op::v1::Reshape>(new_wi, new_const, false);
        }
        NPUW_ASSERT(gather_in);
```

So then you can use it as input to gather unconditionally.

Reply:
Done